stdlib/InteractiveUtils/src/InteractiveUtils.jl

```diff
@@ -104,17 +104,14 @@ function versioninfo(io::IO=stdout; verbose::Bool=false, packages::Bool=false)
     println(io, "  LIBM: ",Base.libm_name)
     println(io, "  LLVM: libLLVM-",Base.libllvm_version," (", Sys.JIT, ", ", Sys.CPU_NAME, ")")
 
-    println(io, "Environment:")
-    for (k,v) in ENV
-        if occursin(r"JULIA", String(k))
-            println(io, "  $(k) = $(v)")
-        end
-    end
-    if verbose
-        for (k,v) in ENV
-            if occursin(r"PATH|FLAG|^TERM$|HOME", String(k))
-                println(io, "  $(k) = $(v)")
-            end
+    env_strs = [String[ "  $(k) = $(v)" for (k,v) in ENV if occursin(r"JULIA", k)];
+                (verbose ?
+                 String[ "  $(k) = $(v)" for (k,v) in ENV if occursin(r"PATH|FLAG|^TERM$|HOME", k)] :
+                 [])]
```

Comment:
Just a suggestion, take it or leave it, but it seems this might be a bit clearer/DRYer as
```julia
env_strs = ["$k = $v" for (k, v) in ENV if occursin("JULIA", k) || (verbose && occursin(r"PATH|FLAG|^TERM$|HOME", k))]
```

Reply:
That occurred to me, but the old code printed all the JULIA variables first, so I figured I'd preserve that.

Reply:
```julia
env_strs = ["$k = $v" for (k, v) in ENV if occursin("JULIA", k)]
if verbose
    append!(env_strs, ["$k = $v" for (k, v) in ENV if occursin(r"PATH|FLAG|^TERM$|HOME", k)])
end
```
?

Anyway, doesn't really matter.